### PR TITLE
protoc-gen-swagger: honor example field of message option

### PR DESCRIPTION
**Pull request by birdayz**
I don't see a possibility to set example values through options. These would be shown by e.g. swagger-ui as example bodies for a POST.

```
message CreatePet {
   option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
    example: {
      value: '{SOME JSON}';
    };
  };

  string name = 1;
}
```

But it also allows setting the example field for nested messages, and the code generator will honor this and just set the nested field to the value.

One thing would be missing: setting examples for fields, i.e. for the string "name"  in the example above. sadly this would need deeper changes, as the field option uses a different data type: JSONSchema and not Schema. Schema contains the example field..
I wonder why this is the case. As far as i have seen, it's possible in Swagger to set the example field even for "primitives".

This patch makes it possible to set examples for messages. While not perfect, it can be a way out of this.
If there's a better possibility to provide examples - without editing the .json manually, please tell me ;)

### CODE REVIEW

**Comment by birdayz** [comment]
Yeah sure, will do that in the next days. Would be great if you could give some input regarding examples for fields. I'd really like to have that, but it would require changes in the .proto files.. and it may not look very nice to have the "example" option in JSONSchema (See https://github.com/grpc-ecosystem/grpc-gateway/blob/master/protoc-gen-swagger/options/openapiv2.proto) as well. I guess there was a good reason to use JSONSchema instead of Schema for fields.

**Comment by johanbrandhorst** [comment]
I don't know about the Schema vs JSONSchema for fields (@ivucica might know) but as for adding an example to the example files, that should be pretty straight forward I hope? It's the easiest way to show the users how to specify these things in their own files, so it should be comprehensive (as you say, include some nested type examples)

**Comment by birdayz** [comment]
Hah yeah it's clear how to add usage examples. Confusing with the "examples" word all over the place.

I was just curious about the "example option field", and if we maybe can work on a possibility to expand this to proto fields as well (and not just for messages, which this patch adds)

File: protoc-gen-swagger/genswagger/template.go [hunk 1]
```diff
@@ -9,6 +9,8 @@ import (
 	"strings"
 	"sync"
 
+	"encoding/json"
```
[the remark this answers is not in the document]
Comment (birdayz) [reply protoc-gen-swagger/genswagger/template.go]: no idea why goimports didn't pick it up. fixed that

**Comment by srenatus** [comment]
🏓 this would really be a nice feature. LMK if there's anything I could help with... 🤔

**Comment by johanbrandhorst** [comment]
If you want to fork this and take over please feel free @srenatus

**Comment by johanbrandhorst** [comment]
As discussed, we're happy to merge this as is and add tests separately. Thanks for your contribution @birdayz!

**Pull request merged**